File: src/main/java/com/neutralplasma/virtusbot/commands/owner/BlackListCommand.java
package com.neutralplasma.virtusbot.commands.owner;

import com.jagrosh.jdautilities.command.CommandEvent;
import com.jagrosh.jdautilities.commons.utils.FinderUtil;
import com.neutralplasma.virtusbot.VirtusBot;
import com.neutralplasma.virtusbot.commands.AdminCommand;
import com.neutralplasma.virtusbot.commands.OwnerCommand;
import com.neutralplasma.virtusbot.handlers.playerLeveling.PlayerData;
import com.neutralplasma.virtusbot.handlers.playerLeveling.PlayerLeveling;
import com.neutralplasma.virtusbot.handlers.playerSettings.PlayerSettingsHandler;
import com.neutralplasma.virtusbot.settings.NewSettingsManager;
import com.neutralplasma.virtusbot.storage.LocaleHandler;
import com.neutralplasma.virtusbot.storage.MySQL;
import com.neutralplasma.virtusbot.storage.PlayerStorage;
import com.neutralplasma.virtusbot.storage.SQL;
import com.neutralplasma.virtusbot.utils.FormatUtil;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.entities.Role;
import net.dv8tion.jda.api.entities.TextChannel;

import java.awt.*;
import java.sql.SQLException;
import java.util.List;

public class BlackListCommand extends OwnerCommand {


    public BlackListCommand(){
        this.name = "blacklist";
        this.help = "Owner blacklist.";
        this.arguments = "<SUBCOMMAND>";
    }


    @Override
    protected void execute(CommandEvent commandEvent) {
        Guild guild = commandEvent.getGuild();
        String[] args = commandEvent.getArgs().split(" ");


        if(args.length > 0){
            if(args[0].equalsIgnoreCase("add")){

                List<Member> members = FinderUtil.findMembers(args[1], guild);
                if (members.isEmpty()) {
                    commandEvent.reply("No user found!");
                } else if (members.size() > 1) {
                    commandEvent.reply(commandEvent.getClient().getWarning() + "You need to specify just 1 user!");
                } else {
                    VirtusBot.getBlackList().addToBlackList(members.get(0).getId());
                    commandEvent.reply("Added to blacklist: " + members.get(0).getId());
                }

            }
            if(args[0].equalsIgnoreCase("remove")){
                List<Member> members = FinderUtil.findMembers(args[1], guild);
                if (members.isEmpty()) {
                    commandEvent.reply("No user found!");
                } else if (members.size() > 1) {
                    commandEvent.reply(commandEvent.getClient().getWarning() + "You need to specify just 1 user!");
                } else {
                    VirtusBot.getBlackList().removeFromBlackList(members.get(0).getId());
                    commandEvent.reply("Removed from blacklist: " + members.get(0).getId());
                }

            }

            if(args[0].equalsIgnoreCase("check")){
                List<Member> members = FinderUtil.findMembers(args[1], guild);
                if (members.isEmpty()) {
                    commandEvent.reply("No user found!");
                } else if (members.size() > 1) {
                    commandEvent.reply(commandEvent.getClient().getWarning() + "You need to specify just 1 user!");
                } else {
                    if(VirtusBot.getBlackList().isBlackListed(members.get(0).getId())){
                        commandEvent.reply("User is blacklisted!");
                    }else{
                        commandEvent.reply("User isn't blacklisted!");
                    }
                }

            }

        }
    }

}

File: src/main/java/com/neutralplasma/virtusbot/commands/owner/BlackListCommand.kt
package com.neutralplasma.virtusbot.commands.owner

import com.jagrosh.jdautilities.command.CommandEvent
import com.jagrosh.jdautilities.commons.utils.FinderUtil
import com.neutralplasma.virtusbot.VirtusBot.blackList
import com.neutralplasma.virtusbot.commands.OwnerCommand

class BlackListCommand : OwnerCommand() {
    override fun execute(commandEvent: CommandEvent) {
        val guild = commandEvent.guild
        val args = commandEvent.args.split(" ".toRegex()).toTypedArray()
        if (args.isNotEmpty()) {
            if (args[0].equals("add", ignoreCase = true)) {
                val members = FinderUtil.findMembers(args[1], guild)
                if (members.isEmpty()) {
                    commandEvent.reply("No user found!")
                } else if (members.size > 1) {
                    var list = "Blacklisted: "
                    for(member in members){
                        blackList.addToBlackList(member.id)
                        list += "${member.id}, "
                    }
                    commandEvent.reply(list)
                } else {
                    blackList.addToBlackList(members[0].id)
                    commandEvent.reply("Added to blacklist: " + members[0].id)
                }
            }
            if (args[0].equals("remove", ignoreCase = true)) {
                val members = FinderUtil.findMembers(args[1], guild)
                if (members.isEmpty()) {
                    commandEvent.reply("No user found!")
                } else if (members.size > 1) {
                    var list = "Removed: "
                    for(member in members){
                        blackList.removeFromBlackList(member.id)
                        list += "${member.id}, "
                    }
                    commandEvent.reply(list)
                } else {
                    blackList.removeFromBlackList(members[0].id)
                    commandEvent.reply("Removed from blacklist: " + members[0].id)
                }
            }
            if (args[0].equals("check", ignoreCase = true)) {
                val members = FinderUtil.findMembers(args[1], guild)
                if (members.isEmpty()) {
                    commandEvent.reply("No user found!")
                } else if (members.size > 1) {

                    var list = "Removed: "
                    for(member in members){
                        if(blackList.isBlackListed(member.id)){
                            list += "${member.id} - Blacklisted, \n"
                        }else{
                            list += "${member.id} - Not blacklisted, \n"
                        }
                    }
                    commandEvent.reply(list)

                } else {
                    if (blackList.isBlackListed(members[0].id)) {
                        commandEvent.reply("User is blacklisted!")
                    } else {
                        commandEvent.reply("User isn't blacklisted!")
                    }
                }
            }
        }
    }

    init {
        name = "blacklist"
        help = "Blacklist user from using the bot."
        arguments = "<add,remove,check>"
    }
}